api/src/main/java/marquez/service/OpenLineageService.java

```diff
@@ -74,12 +75,29 @@ public OpenLineageService(BaseDao baseDao, RunService runService) {
                         event.getJob().getName(),
                         event.getJob().getNamespace(),
                         createJsonArray(event, mapper),
-                        event.getProducer().toString())),
+                        event.getProducer())),
             ForkJoinPool.commonPool());
 
     return CompletableFuture.allOf(marquez, openLineage);
   }
 
+  /**
+   * Try to convert the run id to a UUID. If it isn't a properly formatted UUID, generate one from
+   * the string bytes
+   *
+   * @param run
+   * @return the {@link UUID} for the run
+   */
+  private UUID runUuidFromEvent(LineageEvent.Run run) {
```

Comment:
_Minor_: A runID for OpenLineage should be a `UUID` according to the spec, but assuming this is for backwards compatibility?

Reply:
This was the original logic. I simply reverted the prior commit.
But yeah, this does seem like a good backward compatibility mechanism- the worst case is that the catch block is never executed.